src/Core/Mark.php

```diff
@@ -1,28 +1,24 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace Tiptap\Core;
 
-class Mark
+abstract class Mark
 {
     public static $name;
+    public array $options = [];
 
     public function __construct(array $options = [])
     {
         $this->options = array_merge($this->addOptions(), $options);
     }
 
-    public function addOptions()
+    public function addOptions(): array
     {
-        return [];
+        return [
+            'HTMLAttributes' => [],
+        ];
     }
 
-    public function renderHTML($mark)
-    {
-        return null;
-    }
-
-    public function parseHTML()
-    {
-        return [];
-    }
+    abstract public function renderHTML($mark, array $HTMLAttributes = []): array;
```

Comment:
@hanspagel the `$mark` parameter and `$node` parameter on the `Node` class is unused. Did you have think of any future usecase, where you need to access those values inside the renderHTML method? It's probably for own implementations. Am I write about this?